backend/src/database/users.go

```diff
@@ -167,6 +167,16 @@ func (db *DB) GetUserByUsername(username string) *models.User {
 	return &user
 }
 
+func (db *DB) IsValidUsername(username string) bool {
```

Comment:
Any chance we can rename this? IsValidUsername implies a check to make sure the name doesn't violate any rules (eg. no spaces) whereas this function checks to see if the username has already been taken.

Reply:
Good catch I wasn't using my head there.